src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -3026,6 +3168,23 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
 	case SN_ToScalar: {
 		if (!is_element_type_primitive (fsig->params [0]))
 			return NULL;
+
+		MonoClass *arg_class = mono_class_from_mono_type_internal (fsig->params [0]);
+
+		if (fsig->params [0]->type != MONO_TYPE_GENERICINST) {
+			// This exists to handle the static extension methods for Vector2/3/4, Quaternion, and Plane
+			// which live on System.Numerics.Vector
+
+			arg0_type = MONO_TYPE_R4;
+		}
+
+		int size = mono_class_value_size (arg_class, NULL);
+
+		if (size != 16) {
```

Comment:
Do we need this check here? Was this running for Vector 2/3 before this change?

Reply:
No, the `ToScalar` APIs are new as of this PR.

The blocker here is the same as mentioned above, there needs to be a centralized helper that ensures the handling of Vector2/3 is done correctly to avoid the invalid IR shape otherwise being created.

Reply:
Maybe I am missing something. It looks to me that we had some old implementation here in place (lines 3188 - 3189)

Reply:
Sorry, let me clarify.

`ToScalar` existed for `Vector128`. It's new for `Vector2/3/4`. This is just blocking the handling for Vector2/3 due to the mentioned issue.